back-end/utils/redis-utils/PendingOrders.js

```diff
@@ -88,30 +93,36 @@ const emptyPendingTransactionsListOneCompany = (companyCode, recentPrice) => {
  */
 const emptyPendingTransactionsListAllCompanies = (companyCodes) => {
   return new Promise((resolve, reject) => {
-    companyCodes
-      .forEach((companyCode) => {
+    const emptyAllCompaniesPromises = companyCodes
+      .map((companyCode, id) => {
         const redisKeyQuote = `cachedShares|${companyCode}|quote`;
 
         getAsync(redisKeyQuote)
           .then((quote) => {
-            if (quote) return parseCachedShareQuote(quote);
-            else reject(new Error(`No quote was found for ${companyCode}`));
-          })
-          .then((quoteJSON) => {
-            if (quoteJSON)
-              emptyPendingTransactionsListOneCompany(
-                companyCode,
-                quoteJSON.price
-              );
+            if (quote) {
+              const quoteJSON = parseCachedShareQuote(quote);
+              if (quoteJSON)
+                emptyPendingTransactionsListOneCompany(
+                  companyCode,
+                  quoteJSON.price
+                );
+            } else reject(new Error(`No quote was found for ${companyCode}`));
```

Comment:
Thật ra anh nên check lại cái flow, anh muốn dùng cái function này chỗ nào. Anh nhắn nhắc em lại cái flow nhá! Rồi mình sẽ xử lí cái else này.

Reply:
I don't really understand what you mean, can you explain the details